### PR TITLE
Merge from v2.x

### PR DESCRIPTION
**Pull request by efiring**
This merge from v2.x into master also required a little editing, and it involves many changes, hence this PR.  I resolved conflicts in .travis.yml and lib/matplotlib/tests/test_colors.py.

### CODE REVIEW

**Comment by efiring** [comment]
@tacaswell or @jenshnielsen Only appveyor is running on this--no Travis.  I imagine this means I broke .travis.yml, but I don't see what the problem is.

File: .travis.yml [hunk 1]
```diff
@@ -76,17 +76,14 @@ install:
     pip install --upgrade pip
     pip install --upgrade wheel
     pip install --upgrade setuptools
-  - |
-    # Install dependencies
-    if [ -z "$PRE" ]; then
-        # Install only from travis wheelhouse
-        wheelhouse_pip_install $NUMPY $PANDAS;
+  # Install only from travis wheelhouse
+  - if [ -z "$PRE" ]; then
```
Comment (QuLogic) [inline .travis.yml]: The pipe on the first line is necessary here to get proper multi-line semantics.

Comment (efiring) [inline .travis.yml]: Thank you--I completely missed that. I will put it in now.

File: .travis.yml [hunk 2]
```diff
@@ -77,16 +77,14 @@ install:
     pip install --upgrade wheel
     pip install --upgrade setuptools
   - |
-    # Install dependencies
+  # Install only from travis wheelhouse
```
Comment (QuLogic) [inline .travis.yml]: This line will need to be indented to match.

**Comment by efiring** [comment]
@dopplershift, I had to add `frame_size_can_vary` to `NullMovieWriter` in test_animation.py.  I suspect it could be added to the `AbstractMovieWriter` instead, but I don't understand all this ABC business, so I'm not sure.  I presume there is some reason you didn't add it there.

**Comment by dopplershift** [comment]
@efiring I think `AbstractMovieWriter` was added post 1.5, so no, I didn't leave it off on purpose--there was no option. 😄  Probably would be better on master to move the property to the abstract base class...

**Pull request merged**